tests/common/dualtor/tor_failure_utils.py

```diff
@@ -0,0 +1,124 @@
+"""
+Tor Failure utilities to test switchover and MUX handling during:
+Shutdown all BGP sessions on a ToR
+Shutdown the LinkProber on a ToR
+Blackhole all traffic on a ToR
+Reboot a ToR
+"""
+from tests.common.reboot import *
+import ipaddress
+import pytest
+import logging
+import time
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def shutdown_tor_bgp():
```

Comment:
Will it be simpler to define those utilities as functions instead of fixtures and let the user import them to use?

Reply:
Agree here. Could we make these methods so that the tests can decide when they want to call them? I think it would be easier to measure the traffic disruption this way, since we cannot directly control when a fixture is run.

Reply:
@lolyu I think Lawrence's comment is the right answer. In summary, I see two major benefits of keeping the fixtures:

1. Tests do not have to care for the cleanup/restoration. This also ensures if a test fails, the fixtures cleanup will still be called, instead of a try/catch/finally or cleanup in a test. Keeping tests appear simple.
2. Tests do not have to send every argument to the function. The outer fixture will keep session level fixtures available for inner logic. This will prevent putting args in test definitions and in the function calls. An example in this PR is a `reboot_tor` fixture which keeps localhost fixture available for inner function.